pallets/aux/shift-session-manager/src/lib.rs

```diff
@@ -0,0 +1,280 @@
+                .cloned()
+                .collect(),
+            _ => available_validators[offset..end].to_vec(),
+        };
+
+        session_validators
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    // From construct_runtime macro
+    #![allow(clippy::from_over_into)]
+
+    use super::*;
+    use frame_support::{
+        parameter_types,
+        sp_io::TestExternalities,
+        sp_runtime::{
+            testing::{Header, UintAuthorityId},
+            traits::{BlakeTwo256, ConvertInto, IdentityLookup},
+            Perbill, RuntimeAppPublic,
+        },
+        traits::GenesisBuild,
+        weights::Weight,
+        BasicExternalities,
+    };
+    use sp_core::H256;
+
+    type AccountId = u64;
+
+    type Block = frame_system::mocking::MockBlock<TestRuntime>;
+    type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<TestRuntime>;
+
+    frame_support::construct_runtime! {
+        pub enum TestRuntime where
+            Block = Block,
+            NodeBlock = Block,
+            UncheckedExtrinsic = UncheckedExtrinsic,
+        {
+            System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+            Session: pallet_session::{Pallet},
+        }
+    }
+
+    parameter_types! {
+        pub const BlockHashCount: u64 = 250;
+        pub const MaximumBlockWeight: Weight = 1024;
+        pub const MaximumBlockLength: u32 = 2 * 1024;
+        pub const AvailableBlockRatio: Perbill = Perbill::one();
+    }
+
+    impl frame_system::Config for TestRuntime {
+        type Origin = Origin;
+        type Index = u64;
+        type Call = Call;
+        type BlockNumber = u64;
+        type Hash = H256;
+        type Hashing = BlakeTwo256;
+        type AccountId = AccountId;
+        type Lookup = IdentityLookup<Self::AccountId>;
+        type Header = Header;
+        type Event = ();
+        type BlockHashCount = BlockHashCount;
+        type Version = ();
+        type PalletInfo = PalletInfo;
+        type AccountData = ();
+        type OnNewAccount = ();
+        type OnKilledAccount = ();
+        type BaseCallFilter = frame_support::traits::Everything;
+        type SystemWeightInfo = ();
+        type BlockWeights = ();
+        type BlockLength = ();
+        type DbWeight = ();
+        type SS58Prefix = ();
+        type OnSetCode = ();
+        type MaxConsumers = frame_support::traits::ConstU32<16>;
+    }
+
+    parameter_types! {
+        pub const Period: u64 = 1;
+        pub const Offset: u64 = 0;
+    }
+
+    impl pallet_session::Config for TestRuntime {
+        type Event = ();
+        type ValidatorId = <Self as frame_system::Config>::AccountId;
+        type ValidatorIdOf = ConvertInto;
+        type ShouldEndSession = pallet_session::PeriodicSessions<Period, Offset>;
+        type NextSessionRotation = pallet_session::PeriodicSessions<Period, Offset>;
+        type SessionManager = ();
+        type SessionHandler = TestSessionHandler;
```

Comment:
isn't here () impl?

Reply:
No, the `SessionHandler` trait is not implemented for `()`, hence this `TestSessionHandler`